crates/teloxide/examples/chat_member_updates.rs

```diff
@@ -0,0 +1,84 @@
+use teloxide::{
+    dispatching::Dispatcher,
+    prelude::*,
+    types::{ChatMemberUpdated, ParseMode, Update},
+    utils::html,
+};
+use teloxide_core::adaptors::DefaultParseMode;
+
+/// We need the Bot alias type because we need use the
+/// `ParseMode::Html` by default
+/// (This is optional for this example, use the `parse_mode` method
+/// for set the parse mode in a specific message)
+type Bot = DefaultParseMode<teloxide::Bot>;
+
+#[tokio::main]
+async fn main() -> ResponseResult<()> {
+    pretty_env_logger::init();
+
+    // We need use `ParseMode::Html` by default because the `first_name` or
+    // `username` is formatted from `<a href="tg://user?
+    // id={user_id}>{username}</a>"` to a message with link in the
+    // chat_member_handler function
+    let bot = teloxide::Bot::from_env().parse_mode(ParseMode::Html);
+
+    // We create a handler for our bot
+    let handler = dptree::entry()
+        .inspect(|u: Update| {
+            println!("{u:#?}"); // Print the update to the console with inspect
+                                // method and a closure for debug purposes
+        })
+        .branch(
+            Update::filter_chat_member() // We use filter_chat_member method for filter the updates
+                .filter(|m: ChatMemberUpdated| {
```

Comment:
It's redundant logic 
You can listen left or new chat member by built-in filters
`Update::filter_new_chat_member`
and
`Update::filter_left_chat_member`

Reply:
![Screenshot_2023-08-08-16-24-55-057_com.github.android.jpg](https://github.com/teloxide/teloxide/assets/108593932/1900971e-12a0-4e74-abab-fe00944e4b69)

They had asked me for something like that and I couldn't find the Update::filter_new...

Maybe I'm looking for it wrong 😅

Reply:
@fr0staman note that those filters are on `Message`, not `Update`. Also, I think they are different? Not sure if bots get those messages for all groups...

Reply:
@CrawKatt wouldn't it be better to have a branch here? i.e. something like

```rust
Update::filter_chat_member()
    .branch(
        dptree::filter(|m: ChatMemberUpdated| {
            m.old_chat_member.is_left() && m.new_chat_member.is_present()
        })
        .endpoint(new_chat_member),
    )
    .branch(
        dptree::filter(|m: ChatMemberUpdated| {
            m.old_chat_member.is_present() && m.new_chat_member.is_left()
        })
        .endpoint(left_chat_member),
    );
```

Reply:
That looks good, I'm going to test it in the code and update it. I think it will take me a while because I recently reformatted my PC to migrate from distro. I went from Arch Linux to Fedora

Reply:
I have a question, to test that the example compiles and works correctly, you have to use teloxide from git in the cargo.toml? i made a test project to test the code and for some reason i have a compile error because of a type problem but if i apply the type change (see last issue) teloxide gives compile error because it was expecting the type it was supposed to convert to be able to compile. why is this? i leave some demo images below.

In this case, no problems
![teloxide_repo](https://github.com/teloxide/teloxide/assets/108593932/4dcabadc-7131-4d6f-ad33-da204ff0bb6c)

In this case, i have a compile error because a mismatched types
![trouble_1](https://github.com/teloxide/teloxide/assets/108593932/cd5b8eea-c68a-4e8c-939b-2a90fdc93313)

Reply:
Ok, I'm going to test the example 👍🏻

Reply:
Btw, another tip: you can enclose your screenshots in

```
<details><summary>big screenshot (or any description)</summary>
<p>

(your screenshot (or anything, really) here)

</p>
</details>
```

so that it produces a foldable (& folded by default) thingy:

<details><summary>big screenshot (or any description)</summary>
<p>

(your screenshot (or anything, really) here)

</p>
</details>

That way big stuff like screenshots or particularly long code examples don't use too much of the screen space by default.